kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.java

```diff
@@ -470,6 +470,16 @@ private void refreshForm() {
             refreshFieldsBasedOnInterface(this.selectedNetIfConfig.get());
             refreshFieldsBasedOnSelectedValues();
         }
+
+        // Show read-only dns field when DHCP is selected and there are no
+        // custom DNS entries
+        String configureValue = this.configure.getSelectedItemText();
+        if (configureValue.equals("netIPv6MethodDhcp")
+                && (this.dns.getValue() == null || this.dns.getValue().isEmpty())) {
```

Comment:
`getValue` returns `null` if the field is empty or the parsing fails, so there is no need to call `getValue().isEmpty()`

Reply:
🤔 I tested with only checking for `null` and it didn't work correctly. I'll leave it as-is for now.